File: Backend/src/main/java/org/example/backend/dto/TimeslotDto.java
package org.example.backend.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Value;

import java.io.Serializable;

/**
 * DTO for {@link org.example.backend.domain.Timeslot}
 */
@Data
@NoArgsConstructor
public class TimeslotDto implements Serializable {
    Integer id;
    Short year;
    String day;
    Short hour;
    ClassDto classField;
    TeacherDto teacher;
    SubjectDto subject;
    ClassroomDto classroom;
}

File: Frontend/src/app/dto/timeslot-dto.ts
import {ClassDto} from "./class-dto";
import {TeacherDto} from "./teacher-dto";
import {SubjectDto} from "./subject-dto";
import {ClassroomDto} from "./classroom-dto";

export interface TimeslotDto {
  id : number
  year : number
  day : string
  hour : number
  classField : ClassDto
  teacher : TeacherDto
  subject : SubjectDto
  classroom : ClassroomDto
}
